### DISCUSSION TITLE
How do I create a migration when creating a Django app?

### DISCUSSION THREAD

**Discussion by abe-101** [browniebroke/pypackage-template#817] [Q&A] [answered]
I'm at a loss as to how i should go about creating a migration for Django app that i generated from this template?

**Comment by browniebroke** [comment, marked as answer]
Great question. I've had this issue while trying to contribute to reusable Django apps, it's a real problem.

I thought the answer would be simpler, but while trying it out, I found several issues with the current setup. Basically, you need to:

- Add a `manage.py` at the root of your project. This can be taken from the standard `django-admin startproject` and pointing the default setting file to the test settings: `os.environ.setdefault("DJANGO_SETTINGS_MODULE", "tests.settings")` at the beginning of `main()` function
- Create a `migrations/__init__.py` in your app
- Run `python manage.py makemigrations --skip-checks` and you should see your new migration file. The `--skip-checks` is required because the default settings are failing the checks.

I'll see if I can make it work better out of the box, thanks for the question!

**Reply by browniebroke** [reply]
Latest version of the template was updated here: https://github.com/browniebroke/pypackage-template/pull/818

**Reply by abe-101** [reply]
Thank you so much

**Comment by abe-101** [comment]
when answering yes to is it a django package
it puts this in pyproject.toml
```
django_find_project = false
```
which causes teh tests to fail

**Reply by browniebroke** [reply]
Could you please start a new issue? That sounds like a bug.... Please describe the commands you run (tox/pytest/IDE)

Thanks!
